lsp/nls/src/requests/completion.rs

```diff
@@ -780,6 +792,48 @@ pub fn handle_completion(
     Ok(())
 }
 
+fn handle_import_completion(
+    import: &OsString,
+    params: &CompletionParams,
+) -> io::Result<Vec<CompletionItem>> {
+    debug!("handle import completion");
+
+    let current_file = params
+        .text_document_position
+        .text_document
+        .uri
+        .to_file_path()
+        .unwrap()
+        .canonicalize()?;
+    let mut current_path = current_file.clone();
+    current_path.pop();
+    current_path.push(import);
+    let dir = std::fs::read_dir(&current_path)?;
```

Comment:
Just sort of thinking out loud here (and definitely not a blocker for this PR), but maybe it's worthwhile also looking for completions in the filenames cached in `server.files`? There are some niche cases where that might help (files that have been created but not yet saved) but more importantly it would make this feature testable with the current testing harness (which synthesizes inputs without writing them to disk).

Reply:
True, testing was something I was thinking about trying to implement but couldn't really think of a way without having files on disk. I'll see if I can implement using the server-cached files

Reply:
Got it working
![image](https://github.com/tweag/nickel/assets/89555032/774ee670-605b-4c80-a403-ba0376f3b2d6)
![image](https://github.com/tweag/nickel/assets/89555032/80755b15-73e7-4f8f-be4e-a3566740bfd9)